src/virtualElements.js

```diff
@@ -133,19 +140,30 @@
         },
 
         firstChild: function(node) {
-            if (!isStartComment(node))
+            if (!isStartComment(node)) {
+                if (node.firstChild && isUnmatchedEndComment(node.firstChild)) {
```

Comment:
This code assumes that we've already called getVirtualChildren in these node, but that isn't necessarily true.

Reply:
Interesting, that's certainly true. Is it ever actually valid to have a closing comment as the first child of any element though? Getting the first child of the div in `<!-- ko text abc --><div><!-- /ko --></div>` should really throw an error, right? We know at that point, with no wider context, that your structure is definitely invalid (in a normal applyBinding the full example would throw an error anyway, complaining about the start comment being unmatched, but the specific firstChild call case is interesting standalone to).

If we can take as given that _any_ end comment here is invalid, then this could just throw an error like 'Found closing virtual element as first child of [node]', and not worry about the matching here either way.

Reply:
Right. This could just check `isEndComment(node.firstChild)`

Reply:
Done, and I've added an extra test to try and cover these cases a little better en route too.